Cinema-Reservation-System-Front-End-Client-Part-Internship-project/src/app/authentication/register/register.component.ts

```diff
@@ -0,0 +1,84 @@
+import { Component } from '@angular/core';
+import { ButtonComponent } from '../../shared/button/button.component';
+import { InputFieldComponent } from '../../shared/input-field/input-field.component';
+import {
+  FormBuilder,
+  FormGroup,
+  ReactiveFormsModule,
+  Validators,
+} from '@angular/forms';
+import { AuthenticationService } from '../shared/authentication.service';
+
+@Component({
+  selector: 'app-register',
+  standalone: true,
+  imports: [ButtonComponent, InputFieldComponent, ReactiveFormsModule],
+  templateUrl: './register.component.html',
+  styleUrl: './register.component.css',
+})
+export class RegisterComponent {
+  buttonLabel: string = 'Register';
+  placeHolders = {
+    firstName: 'First Name',
+    lastName: 'Last Name',
+    email: 'Email',
+    password: 'Password',
+    confirmPassword: 'Confirm Password',
+  };
+  constructor(
+    private fb: FormBuilder,
+    private authService: AuthenticationService,
+  ) {}
+
+  registerForm: FormGroup = new FormGroup({});
+  formSubmitted = false;
+
+  ngOnInit() {
+    this.registerForm = this.fb.group({
+      firstName: [
+        '',
+        Validators.compose([Validators.required, Validators.minLength(2)]),
+      ],
+      lastName: [
+        '',
+        Validators.compose([Validators.required, Validators.minLength(2)]),
+      ],
+      email: ['', Validators.compose([Validators.required, Validators.email])],
+      password: [
+        '',
+        Validators.compose([Validators.required, Validators.minLength(5)]),
+      ],
+      confirmPassword: [
+        '',
+        Validators.compose([Validators.required, Validators.minLength(5)]),
```

Comment:
We can remove m,inLength validation here, we only need to check that this fieeld is filled and is equal to the password. You can add your own validation function to check the equality of two fields.